src/ApiService/ApiService/Functions/QueueFileChanges.cs

```diff
@@ -56,6 +94,51 @@ public class QueueFileChanges {
         var path = string.Join('/', parts.Skip(1));
 
         log.Info($"file added : {container:Tag:Container} - {path:Tag:Path}");
-        await _notificationOperations.NewFiles(Container.Parse(container), path, failTaskOnTransientError);
+        await _notificationOperations.NewFiles(Container.Parse(container), path, isLastRetryAttempt);
     }
+
+    private async Async.Task RequeueMessage(string msg) {
+        var json = JsonNode.Parse(msg);
+
+        // Messages that are 'manually' requeued by us as opposed to being requeued by the azure functions runtime
+        // are treated as new every single time. Thus, the dequeue count doesn't persist.
+        // That's why we need to keep track of it ourselves.
+        var newCustomDequeueCount = 0;
+        if (json!["data"]?["customDequeueCount"] != null) {
+            newCustomDequeueCount = json["data"]!["customDequeueCount"]!.GetValue<int>() + 1;
+        }
+
+        json!["data"]!["customDequeueCount"] = newCustomDequeueCount;
+
+        if (newCustomDequeueCount > 6) {
+            _log.Warning($"Message retried for over a week with no success: {msg}");
+        }
+
+        await _context.Queue.QueueObject(
+            QueueFileChangesPoisonQueueName,
+            json,
+            StorageType.Config,
+            CalculateExponentialBackoff(newCustomDequeueCount))
+            .IgnoreResult();
+    }
+
+    // Possible return values:
+    // 1 - 5 mins
+    // 2 - 25 mins
+    // 3 - 2 hours 5 mins
+    // 4 - 10 hours 25 mins
+    // 5 - 2 days 4 hours 5 mins
+    // >6 - 7 days +/- 1 day of random variance so all the messages don't dequeue at the same time
+    private static TimeSpan CalculateExponentialBackoff(int dequeueCount) {
+        var numMinutes = Math.Pow(5, dequeueCount);
+        var backoff = TimeSpan.FromMinutes(numMinutes);
+        var maxBackoff = MaxTtl();
+        return backoff < maxBackoff ? backoff : maxBackoff;
+    }
+
+    private static TimeSpan MaxTtl() {
+        var variance = new Random().Next(0, 48);
```

Comment:
Use the 'shared' random number generator instead of instantiating a new one:

```suggestion
        var variance = Random.Shared.Next(0, 48);
```